DEVELOPMENT.md

```diff
@@ -1,47 +1,44 @@
 # Developing the Tekton website
 
 - [Developing the Tekton website](#developing-the-tekton-website)
-  - [Running Locally](#running-locally)
-  - [tekton.dev](#tektondev)
+- [Running Locally](#running-locally)
+- [tekton.dev](#tektondev)
 
-## Running Locally
-
-
-The [tekton.dev](#tekton.dev) is deployed on Netlify. Netlify will invoke
-[the Makefile](.Makefile) to build the website.
-
-This is configured in:
-
-* [netlify.toml](netlify.toml)
-* [runtime.txt](runtime.txt) (the python runtime)
-* [requirements.txt](requirements.txt) (the python requirements)
-
-To run it locally you will need to install:
+## Dependencies
 
-* [Hugo](https://gohugo.io/) [version 0.53](netlify.toml)
-* `npm` to install Hugo's requirements:
-  * `npm install` (in directory with [package.json](package.json))
+* `python3` [installation link](https://www.python.org/downloads/) 
+* `hugo (EXTENDED VERSION)` [installation link](https://github.com/gohugoio/hugo/releases)
+* `pip` [installation link](https://pip.pypa.io/en/stable/installing/)
+* `npm v6.14.5` [installation link](https://nodejs.org/en/)
+* `node v14.3.0` [installation link](https://nodejs.org/en/)
+* `netlify cli` [installation link](https://cli.netlify.com/getting-started)
+* `netlify account` [sign up](https://app.netlify.com/)
 
-Then you can run [hugo commands](https://gohugo.io/getting-started/usage/) such as:
+## Running Locally
 
+Step 1
 ```bash
-hugo server
+git clone https://github.com/tektoncd/website && cd website
 ```
-
-To run the [Makefile](Makefile) you will need more dependencies and flags:
-
-* `python` aliased to python3.7 - [sync.py](sync/sync.py) is running `python` but assumes this is python3 (configured in [runtime.txt](runtime.txt))
-* `pip`, to install python requirements:
-  * Install with `pip install -r requirements.txt`
-
-To run the makefile locally, you'll have to set the env var:
-
-* [`DEPLOY_PRIME_URL`](https://docs.netlify.com/configure-builds/environment-variables/) to the path to your website files
-
-For example:
-
+Step 2
+```bash
+node i
+```
+Step 3
+```bash
+pip install -r requirements.txt
+```
+Step 4
+```bash
+alias python=python3
+```
+Step 5
+```bash
+python sync/sync.PY
+```
+Step 6
 ```bash
-DEPLOY_PRIME_URL=localhost make preview-build
+netlify dev
```

Comment:
won't `netlify dev` run the sync for us?

running sync explicitly should only be needed if running `hugo server` directly as far as I understand

Reply:
Yes, but if you install hugo and just try to run it to see if it works. It looks like hugo it is failing for some random reason.  It is pretty confusing, but considering that we are guiding the user step by step. This makes alot of sense to change it.

Reply:
Actually, idk. No, it doesnt work that way. If you dont sync before doing netlify dev you get this error. ```bash Building sites … ERROR 2020/06/04 18:38:20 render of "section" failed: execute of template failed: template: docs/list.html:8:9: executing "docs/list.html" at <partial "navbar-docs.html" .>: error calling partial: execute of template failed: template: partials/navbar-docs.html:9:15: executing "partials/navbar-docs.html" at <partial "navbar-version-selector-docs.html" .>: error calling partial: "/home/nap/Workplace/go/dev/tekton-website/layouts/partials/navbar-version-selector-docs.html:2:38": execute of template failed: template: partials/navbar-version-selector-docs.html:2:38: executing "partials/navbar-version-selector-docs.html" at <fingerprint>: error calling fingerprint: <nil> can not be transformed
ERROR 2020/06/04 18:38:20 render of "section" failed: execute of template failed: template: docs/list.html:8:9: executing "docs/list.html" at <partial "navbar-docs.html" .>: error calling partial: execute of template failed: template: partials/navbar-docs.html:9:15: executing "partials/navbar-docs.html" at <partial "navbar-version-selector-docs.html" .>: error calling partial: "/home/nap/Workplace/go/dev/tekton-website/layouts/partials/navbar-version-selector-docs.html:2:38": execute of template failed: template: partials/navbar-version-selector-docs.html:2:38: executing "partials/navbar-version-selector-docs.html" at <fingerprint>: error calling fingerprint: <nil> can not be transformed
ERROR 2020/06/04 18:38:20 render of "section" failed: execute of template failed: template: docs/list.html:8:9: executing "docs/list.html" at <partial "navbar-docs.html" .>: error calling partial: execute of template failed: template: partials/navbar-docs.html:9:15: executing "partials/navbar-docs.html" at <partial "navbar-version-selector-docs.html" .>: error calling partial: "/home/nap/Workplace/go/dev/tekton-website/layouts/partials/navbar-version-selector-docs.html:2:38": execute of template failed: template: partials/navbar-version-selector-docs.html:2:38: executing "partials/navbar-version-selector-docs.html" at <fingerprint>: error calling fingerprint: <nil> can not be transformed
ERROR 2020/06/04 18:38:20 render of "section" failed: execute of template failed: template: docs/list.html:8:9: executing "docs/list.html" at <partial "navbar-docs.html" .>: error calling partial: execute of template failed: template: partials/navbar-docs.html:9:15: executing "partials/navbar-docs.html" at <partial "navbar-version-selector-docs.html" .>: error calling partial: "/home/nap/Workplace/go/dev/tekton-website/layouts/partials/navbar-version-selector-docs.html:2:38": execute of template failed: template: partials/navbar-version-selector-docs.html:2:38: executing "partials/navbar-version-selector-docs.html" at <fingerprint>: error calling fingerprint: <nil> can not be transformed
Built in 1320 ms
Error: Error building site: failed to render pages: render of "section" failed: execute of template failed: template: docs/list.html:8:9: executing "docs/list.html" at <partial "navbar-docs.html" .>: error calling partial: execute of template failed: template: partials/navbar-docs.html:9:15: executing "partials/navbar-docs.html" at <partial "navbar-version-selector-docs.html" .>: error calling partial: "/home/nap/Workplace/go/dev/tekton-website/layouts/partials/navbar-version-selector-docs.html:2:38": execute of template failed: template: partials/navbar-version-selector-docs.html:2:38: executing "partials/navbar-version-selector-docs.html" at <fingerprint>: error calling fingerprint: <nil> can not be transformed
◈ "hugo server -w" exited with code 255. Shutting down Netlify Dev server```

Reply:
This is reference to this issue. I dont know why this bug happens. I will see if I can fix it. https://github.com/tektoncd/website/issues/3

Reply:
I also believe opening an issue around the sync script should be made. It does more than just 'sync the docs' at this point and there should be a basic test for checking for broken links. I feel like a design doc is definitely needed. Hopefully, I can get to it pretty soon.